app/services/send_channel_invitation_service.rb

```diff
@@ -0,0 +1,25 @@
+class SendChannelInvitationService
+  attr_accessor :invitation_usernames, :current_user, :chat_channel
+
+  def initialize(invitation_usernames, current_user, chat_channel)
+    @invitation_usernames = invitation_usernames
+    @current_user = current_user
+    @chat_channel=chat_channel
+  end
+
+  def send_invitations
+    usernames = invitation_usernames.split(",").map do |username|
+      username.strip.delete("@")
+    end
+    if usernames.present?
+      users = User.where(username: usernames)
+      invitations_sent = chat_channel.invite_users(users: users, membership_role: "member", inviter: current_user)
+      message = if invitations_sent.zero?
+                  "No invitations sent. Check for username typos."
```

Comment:
**suggestion (non-blocking):** Maybe in a future PR, we can add some form of auto-complete on username, at least for mutual follows.